docs/DeveloperGuide.md

```diff
@@ -369,6 +376,23 @@ and only describes how regular, properly formatted save files are loaded in
 The Storage class also saves and loads the list of user's available ingredients to another .txt save file.
 The mechanisms with which it does so is highly similar to the 2 sequence diagrams above.
 
+### Exception Class
+There is a main YMFCException class that extends the Exception class.
+There are then 4 child exception classes that inherit from the YMFCException class.
+* EmptyListException
+  * This exception is thrown mainly in the Parser class whenever the user inputs a command that would not work 
+  with an empty list of recipes or ingredients
+    * e.g. When the user inputs the edit command when the recipe cookbook is empty and hence has to recipes to edit
```

Comment:
This line is a bit confusing, should we rephrase this a bit?

Reply:
oh my bad it should be "no recipes to edit" and not "to recipes to edit".

Reply:
I think this could do: "When the user inputs the edit command but the recipe cookbook is empty, hence no recipes available to be edited"